src/App.tsx

```diff
@@ -15,37 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+type Good = string;
+
+enum SortType {
+  NONE = '',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+const GoodList: React.FC<{ goods: Good[] }> = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li key={good} data-cy="Good">
+        {good}
+      </li>
+    ))}
+  </ul>
+);
+
+function getPreparedGoods(
+  goods: Good[],
+  { sortField, isReversed }: { sortField: SortType; isReversed: boolean },
+) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.ALPHABET,
+          })}
+          onClick={() => {
+            setSortField(SortType.ALPHABET);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortType.LENGTH,
+          })}
+          onClick={() => setSortField(SortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
           Reverse
         </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(SortType.NONE);
+              setIsReversed(false);
+            }}
```

Comment:
Move this logic from jsx to the helper function and use it here
```suggestion
            onClick={reset}
```